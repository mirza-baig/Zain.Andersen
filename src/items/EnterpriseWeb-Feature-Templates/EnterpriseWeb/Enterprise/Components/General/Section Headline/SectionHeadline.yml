---
ID: "c274b7d7-b916-4f82-b3b1-237a55239f89"
Parent: "aef6fd66-45ff-4a2b-92eb-f39822c4240c"
Template: "ab86861a-6030-46c5-b394-e8f99e8b87db"
Path: /sitecore/templates/Feature/EnterpriseWeb/Enterprise/Components/General/Section Headline/SectionHeadline
SharedFields:
- ID: "06d5295c-ed2f-4a54-9bf2-26228d113318"
  Hint: __Icon
  Value: Office/32x32/document_header.png
- ID: "12c33f3f-86c5-43a5-aeb4-5598cec45116"
  Hint: __Base template
  Value: |
    {B61A1D20-9EE5-4CCB-9E0D-24A97E58D293}
    {5FB8ACDF-E52A-4EE9-A3D9-83CD1C5C5CB1}
    {3AA7D5FD-915F-4286-AA99-5020342711BD}
- ID: "f7d48a55-2158-4f02-9356-756654404f73"
  Hint: __Standard values
  Value: "{6AFAED1F-E770-4D8C-AA57-DE4C01C47D9F}"
Languages:
- Language: en
  Fields:
  - ID: "b5e02ad9-d56f-4c41-a065-a133db87bdeb"
    Hint: __Display name
    Value: Section Headline
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20221114T070345Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\Admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\Admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "a0c6cd9e-54a5-455a-954a-c9e5f4d475c3"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230106T120856Z
