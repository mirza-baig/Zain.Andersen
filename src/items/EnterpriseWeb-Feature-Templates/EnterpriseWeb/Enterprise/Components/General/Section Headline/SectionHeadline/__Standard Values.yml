---
ID: "6afaed1f-e770-4d8c-aa57-de4c01c47d9f"
Parent: "c274b7d7-b916-4f82-b3b1-237a55239f89"
Template: "c274b7d7-b916-4f82-b3b1-237a55239f89"
Path: /sitecore/templates/Feature/EnterpriseWeb/Enterprise/Components/General/Section Headline/SectionHeadline/__Standard Values
SharedFields:
- ID: "32a0a4a0-2c24-4894-8629-4cd02476348e"
  Hint: headlineLevel
  Value: "{0F556F3A-BBAD-4AA5-952D-B79003B39CD6}"
- ID: "b03569b1-1534-43f2-8c83-bd064b7d782c"
  Hint: __Renderers
  Value: "{B76947D7-B58E-432D-8D54-8D522674B434}"
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20221114T070506Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\Admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\Admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "775edf98-9d24-4c17-974c-602edebd0913"
    - ID: "b2a74f5b-05db-4e79-8ce9-328512c478cf"
      Hint: headlineText
      Value: $name
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230106T120842Z
