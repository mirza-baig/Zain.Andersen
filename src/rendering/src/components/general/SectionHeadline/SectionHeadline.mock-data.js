const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'SectionHeadline',
  },
  fields: {
    underlineStyle: {
      id: 'a2095115-1185-497b-ae4b-34df594c8312',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Underline-Style/Dark',
      name: 'Dark',
      displayName: 'Dark',
      fields: {
        Value: {
          value: 'dark',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    theme: {
      value: '',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'Section Headline Dark Center',
    },
    alignment: {
      id: 'ed4b79ff-078b-47e0-8374-d30cd00224c9',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Alignment/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventZone: {
      value: '',
    },
    eventType: {
      value: '',
    },
  },
};

export default defaultData;
