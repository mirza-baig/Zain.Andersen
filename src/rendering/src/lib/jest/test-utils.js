// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
