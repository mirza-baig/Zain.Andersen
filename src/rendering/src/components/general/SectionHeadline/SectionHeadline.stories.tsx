import { Story, Meta } from '@storybook/react';

// Local
import SectionHeadline, { SectionHeadlineProps } from './SectionHeadline';
import defaultData from './SectionHeadline.mock-data';

export default {
  title: 'General/SectionHeadline',
  component: SectionHeadline,
} as Meta;

const Template: Story<SectionHeadlineProps> = (props) => <SectionHeadline {...props} />;

export const Default = Template.bind({});
Default.args = defaultData;
