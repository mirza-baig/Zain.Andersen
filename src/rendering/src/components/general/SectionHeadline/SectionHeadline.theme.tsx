import classNames from 'classnames';
import { ThemeFile, ThemeName } from 'lib/context/ThemeContext';
import { BorderStyle, TextAlignment } from './SectionHeadline';

const awHeadlineClasses = 'text-s md:text-m font-heavy pb-xs mb-s';

const rbaHeadlineClasses = 'text-m md:text-l font-extra-light pb-xs mb-s';

const getBorderClasses = (theme: ThemeName, borderStyle: BorderStyle) => {
  if (borderStyle !== 'default') {
    let borderClasses = theme === 'aw' ? 'border-b-6 ' : 'border-b-3 ';

    switch (borderStyle) {
      case 'dark':
        borderClasses += theme === 'aw' ? 'border-primary' : 'border-secondary';
        break;
      case 'light':
        borderClasses += theme === 'aw' ? 'border-light-gray' : 'border-gray';
        break;
      default:
        break;
    }

    return borderClasses;
  }
  return '';
};

const getDynamicStyles = (
  theme: ThemeName,
  alignment: TextAlignment,
  borderStyle: BorderStyle
): string => {
  return classNames(`text-${alignment}`, getBorderClasses(theme, borderStyle));
};

export const SectionHeadlineTheme = (
  alignment: TextAlignment,
  borderStyle: BorderStyle
): ThemeFile => {
  return {
    aw: {
      classes: {
        headlineContainer: classNames(
          awHeadlineClasses,
          getDynamicStyles('aw', alignment, borderStyle)
        ),
      },
    },
    rba: {
      classes: {
        headlineContainer: classNames(
          rbaHeadlineClasses,
          getDynamicStyles('rba', alignment, borderStyle)
        ),
      },
    },
  };
};
