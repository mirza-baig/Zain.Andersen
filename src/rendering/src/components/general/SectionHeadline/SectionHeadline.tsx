import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { Component } from 'src/helpers/Component';
import { Headline } from 'src/helpers/Headline';
import { getEnum } from 'lib/utils';
import { useTheme } from 'lib/context/ThemeContext';
import { SectionHeadlineTheme } from './SectionHeadline.theme';

export type SectionHeadlineProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.SectionHeadline.SectionHeadline;

export type TextAlignment = 'left' | 'center';
export type BorderStyle = 'default' | 'dark' | 'light';

const SectionHeadline = (props: SectionHeadlineProps): JSX.Element => {
  const alignment = getEnum<TextAlignment>(props.fields?.alignment) || 'left';
  const borderStyle = getEnum<BorderStyle>(props.fields?.underlineStyle) || 'default';

  const { themeData } = useTheme(SectionHeadlineTheme(alignment, borderStyle));

  return (
    <Component variant="lg" dataComponent="general/sectionheadline" {...props}>
      <div className="col-span-12">
        <Headline classes={themeData.classes.headlineContainer} {...props} />
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<SectionHeadlineProps>(SectionHeadline);
