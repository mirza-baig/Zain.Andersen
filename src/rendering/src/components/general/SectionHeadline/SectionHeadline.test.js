// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import SectionHeadline from './SectionHeadline';
import defaultData from './SectionHeadline.mock-data';

it('renders correctly', async () => {
  const component = snapshot(SectionHeadline, { componentProps: defaultData });
  await hasDataComponent(component, 'general/sectionheadline');
});
