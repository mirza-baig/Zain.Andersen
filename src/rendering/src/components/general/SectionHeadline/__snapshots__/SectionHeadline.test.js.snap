// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/sectionheadline"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <div
          class="items-top font-sans [&_a:hover]:underline text-s md:text-m font-heavy pb-xs mb-s text-center border-b-6 border-primary"
        >
          <h2>
            Section Headline Dark Center
          </h2>
        </div>
      </div>
    </div>
  </div>
</section>
`;
